### PR TITLE
fix: Validate the type of configmap before loading parameters. Fixes #7312

### LINKED ISSUE #7312: `LabelValueTypeConfigMapParameter` is unused, why?
Why is this new variable unused?

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixes #7312.

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>

Don't bother creating a PR until you've done this:

* [x] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

File: workflow/common/configmap.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package util
+package common
```
Comment (terrytangyuan) [inline workflow/common/configmap.go:1]: Moving this file to common to avoid cyclic dependencies

File: workflow/common/configmap.go [hunk 2]
```diff
@@ -18,6 +18,11 @@ func GetConfigMapValue(configMapInformer cache.SharedIndexInformer, namespace, n
 		if !ok {
 			return "", fmt.Errorf("unable to convert object %s to configmap when syncing ConfigMaps", name)
 		}
+		if cmType := cm.Labels[LabelKeyConfigMapType]; cmType != LabelValueTypeConfigMapParameter {
+			return "", fmt.Errorf(
+				"ConfigMap '%s' needs to have the label %s: %s for parameters loading",
```
Comment (simster7) [inline workflow/common/configmap.go:23]:
```suggestion
				"ConfigMap '%s' needs to have the label %s: %s to load parameters",
```

**Review by simster7 — APPROVED** [review]
Minor change

File: workflow/common/configmap.go (hunk 2 above)
Comment (terrytangyuan) [reply workflow/common/configmap.go]: Thanks!

**Pull request merged**